File: DevExpress-Demo/DevExpress.ProductsDemo.Wpf/VB/DevExpress.ProductsDemo.Wpf/Controls/PrintViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Xpf.Printing

Namespace ProductsDemo

    Public Class PrintViewModel
        Inherits ViewModelBase

        Public Overridable Property PrintableControlLink As PrintableControlLink

        Public Sub UpdatePrintableControlLink()
            If PrintingService.PrintableControlLink IsNot Nothing Then
                PrintableControlLink = PrintingService.PrintableControlLink
                PrintableControlLink.CreateDocument(True)
            End If
        End Sub
    End Class

    Public Module PrintingService

        Public PrintableControlLink As PrintableControlLink

        Public ReadOnly Property HasPrinting As Boolean
            Get
                Return PrintableControlLink IsNot Nothing
            End Get
        End Property
    End Module
End Namespace

File: DevExpress-Demo/DevExpress.ProductsDemo.Wpf/CS/DevExpress.ProductsDemo.Wpf/Controls/PrintViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Xpf.Printing;
using System;
namespace ProductsDemo {
    public class PrintViewModel : ViewModelBase {        
        public virtual PrintableControlLink PrintableControlLink { get; set; }

        public void UpdatePrintableControlLink() {
            if(PrintingService.PrintableControlLink != null) {
                PrintableControlLink = PrintingService.PrintableControlLink;
                PrintableControlLink.CreateDocument(true);
            }
        }
    }
    public static class PrintingService {
        public static PrintableControlLink PrintableControlLink;

        public static bool HasPrinting {
            get {
                return PrintableControlLink != null;
            }
        }
    }
}
